[team-01][쑤, Isaac] ViewModel 구조화, 첫 번째 화면 구성

okstring opened the pull request:
![스크린샷 2021-04-22 오전 11 13 03](https://user-images.githubusercontent.com/62657991/115645668-b6d9a380-a35b-11eb-8574-315f110fa3f4.png)

### 스텝별 작업 목록

- Alamofire 사용해서 네트워킹 및 JSON Decoding
- CollectionView를 활용해 메인 화면 구성
  - 구성을 잡고 가기 위해 먼저 First section만 구성
- CocoaPod을 사용해 의존성 도구 활용
  - `Toaster`, `Alamofire`
- Combine을 활용해 ViewModel - CollectionView 간 binding 
- Caching 을 사용해 FileManager를 사용해서 캐싱 처리 예정



### 학습 키워드

- `Alamafire`
- `Toaster`
- `Combine`
- `FileManager`
- `Cache`
- `CollectionView`



### 고민과 해결(질문거리)

- 저희는 이번 미션을 진행하면서 가장 염두해두고 있는 부분은 View의 구현도 중요하지만 ViewModel, Clean Architecture의 구조를 제대로 알고 싶어 중간 PR을 일찍 보내드리게 됐습니다. 특히나 ViewModel, Binding 부분의 미숙함, UseCase와 ViewModel의 제대로 된 역할을 아직까지 명확하게 알고 있지 않아서 이런  부분도 의심을 해가며 코드를 작성하게 됐습니다. 아직 구조만 짰을 뿐 볼륨이 커지지 않은 상태라 각각의 계층에서 제대로 된 역할을 가질 수 없는 것 같기도 하지만 데이터의 흐름에 대해서도 고민을 했었고 전반적인 흐름에대해 조언을 요청드립니다 :)
- 요구사항에서는 `셀을 표기할 때 이미 다운로드된 이미지가 있으면 표시하고, 새로운 파일이 다운로드 완료되면 해당 이미지를 테이블뷰 셀에 뒤늦게(lazy) 표시한다.` 라고 명시가 되어있는데 lazy하게 표시하기 위해서 이미지를 lazy하게 불러오는 것이 저희가 작성한 흐름과 비교해서 맞는지도 의구심이 들고 있습니다. 
- Error처리를 Network 단에서부터 ViewModel까지 계속해서 가지고 오고 있습니다. 보통의 에러처리와 비교해서 이렇게 긴 흐름을 가지는 Error도 맞는지 궁금했고 Error 처리를 하게되면 저희가 작성한 것 처럼 `ErrorHandler`를 통해서 화면에 일단 표시해두고 있는데 이런 흐름도 맞는지 궁금합니다.



### 부연 설명

- [MVVM, Clean Architecture 패턴에 대해 참고한 자료](https://tech.olx.com/clean-architecture-and-mvvm-on-ios-c9d167d9f5b3)

okstring committed:
Make project init

lenaios committed:
main 스토리보드에 collection view 추가

okstring committed:
Make: HeaderCollectionReusableView

okstring committed:
Make: item label layout 정의

okstring committed:
Define: SidedishItem, Category(Entity)

okstring committed:
Set: DetailViewController nabiagtionBar appear 처리

okstring committed:
Implement: viewModel property 변경 시 reload

okstring committed:
Implement: 대략적인 MVVM 흐름 구성

okstring committed:
Implement: HeaderViewModel 구현, 화면 표시

okstring committed:
Simple: countDescriptionTemplate제거, attributedText 설정

okstring committed:
removed file that shouldn't be tracked

okstring committed:
Implement Toast

okstring committed:
Implement: MVVM 구조 설계, Alamofire, Kingfisher, Toaster 설치

okstring committed:
Implement Error handle

okstring committed:
Modify: networking 시 제네릭으로 변경

lenaios committed:
Solve conflict

lenaios committed:
주석 제거

lenaios committed:
Badge layout 추가, fix original price

okstring committed:
필요없는 파일 및 주석 제거

okstring committed:
Remove .DS_Store

okstring committed:
Apply .gitignore

okstring committed:
누락된 image 흐름 추가

okstring commented:
Image를 받아오는 흐름이 브랜치를 합치는 과정에서 유실된 것 같아 코드 추가 후 푸시드립니다 😅

GangWoon commented:
> 저희는 이번 미션을 진행하면서 가장 염두해두고 있는 부분은 View의 구현도 중요하지만 ViewModel, Clean Architecture의 구조를 제대로 알고 싶어 중간 PR을 일찍 보내드리게 됐습니다. 특히나 ViewModel, Binding 부분의 미숙함, UseCase와 ViewModel의 제대로 된 역할을 아직까지 명확하게 알고 있지 않아서 이런 부분도 의심을 해가며 코드를 작성하게 됐습니다. 아직 구조만 짰을 뿐 볼륨이 커지지 않은 상태라 각각의 계층에서 제대로 된 역할을 가질 수 없는 것 같기도 하지만 데이터의 흐름에 대해서도 고민을 했었고 전반적인 흐름에대해 조언을 요청드립니다 :)


![Screen Shot 2021-04-23 at 1 49 21 AM](https://user-images.githubusercontent.com/48466830/115753962-b7635000-a3d6-11eb-900d-bc8451d54eb4.png)
아래 그림을 보면서 공유해주신 링크를 다시 한번 읽어보셨으면 좋겠어요. 추후에 자세한 내용은 대화를 통해서 가이드 드리겠습니다.

<br>
<br>

> 요구사항에서는 셀을 표기할 때 이미 다운로드된 이미지가 있으면 표시하고, 새로운 파일이 다운로드 완료되면 해당 이미지를 테이블뷰 셀에 뒤늦게(lazy) 표시한다. 라고 명시가 되어있는데 lazy하게 표시하기 위해서 이미지를 lazy하게 불러오는 것이 저희가 작성한 흐름과 비교해서 맞는지도 의구심이 들고 있습니다.


"해당 이미지를 테이블뷰 셀에 뒤늦게(lazy) 표시한다. "
저화질 사진 보여주고 그 다음에 고화질 사진으로 갈아친다고 생각하시면 이해에 도움이 될까요?
이미지는 다른 리소스에 비해 사이즈가 크기 때문에 이미지를 전부 다운 받고 리스트를 갱신하기엔 제한적인 UX입니다.
<br>
<br>


> Error처리를 Network 단에서부터 ViewModel까지 계속해서 가지고 오고 있습니다. 보통의 에러처리와 비교해서 이렇게 긴 흐름을 가지는 Error도 맞는지 궁금했고 Error 처리를 하게되면 저희가 작성한 것 처럼 ErrorHandler를 통해서 화면에 일단 표시해두고 있는데 이런 흐름도 맞는지 궁금합니다.

에러를 사용자에게 표시할 수 있는 순간에 보여주는게 가장 좋습니다. 그 흐름이 길이는 중요하지 않습니다. 
에러라는 단어를 생각해보시면 프로그래머가 원치 않는 순간이란걸 확인하실 수 있습니다.
ErrorHandler를 사용해서 화면에 표시해서 사용자에게 보여주는건 좋은 흐름 같습니다.


---- 
<br>
<br>

구조에 대한 피드백은 토요일에 이어서 드리도록 하겠습니다.

GangWoon commented on Sidedish/Sidedish/Entity/SidedishItem.swift:
```diff
@@ -0,0 +1,32 @@
+    var detailHash: String
+    var image: String
+    var alt: String
+    var deliveryType: [String]?
```
배열자체가 옵셔널인 이유가 있을까요?

lenaios replied:
API 응답 데이터에 deliveryType 값이 없는 경우도 있기 때문에, decoding시 파싱 실패를 피하고자 옵셔널로 지정했습니다.
혹시 다른 방법이 있을까요?🤔

GangWoon replied:
데이터가 없는 경우와 배열이 없는 경우가 동일할까요?

okstring replied:
API 상에서 badge에 값이 있는 경우 `"badge": [ "론칭특가" ]`로 표시되고
badge가 없는 경우 해당 key가 아예 없어서 Optional로 처리하게 됐습니다

만약에 해당 Key와 배열은 있지만 값이 안에 없다면 타입을 옵셔널 처리하는 방식으로 하도록 하겠습니다!

GangWoon commented on Sidedish/Sidedish/Entity/SidedishOfCategory.swift:
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+struct SidedishOfCategory: Codable {
+    var statusCode: Int
```
statusCode를 사용하시지 않는데 프로퍼티로 작성하신 이유가 있을까요?

lenaios replied:
statusCode를 내부 로직에서 사용하고 있지는 않으나, 위와 마찬가지로 API 응답 형태에 따라 decoding 하기 위해 작성했던 값입니다!

GangWoon replied:
statusCode를 저장하지 않고도 가능한 작업이지 않나요?

GangWoon commented on Sidedish/Sidedish/Service/Networking.swift:
```diff
@@ -0,0 +1,34 @@
+    func downloadImage(from url: URL, completion: @escaping ((Data) -> ()) ) {
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
+            guard let data = data, error == nil else {
+                print(error!.localizedDescription)
```
불필요한 프린트 문은 제거해주세요~

okstring replied:
넵 주의하도록 하겠습니다!
에러에 대한 부분을 저희도 인지하도록 하기 위해서 print대신 NSLog로 바꿔서 표시해보려하는데 이런 경우에 알맞게 쓰이는 것이 맞을까요?

GangWoon replied:
디버깅 할 때 사용하실 용도라면 디버깅 옵션에서만 표시할 수 있도록 변경하시는게 좋을 거 같아요

GangWoon replied:
추가적으로 강제언래핑은 앱을 꺼지게만들 수 있는 위험한 방법입니다. 강제언래핑을 줄여보는건 어떨까요?

okstring replied:
감사합니다! Debug 모드에서만 출력이 되도록 NSLog를 사용하도록 수정했습니다
 + 강제 언래핑도 수정했습니다!

GangWoon commented on Sidedish/Sidedish/Service/Networking.swift:
```diff
@@ -0,0 +1,34 @@
+    }
+    
+    func downloadImage(from url: URL, completion: @escaping ((Data) -> ()) ) {
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
```
사용하지 않는 값은 _으로 주석 처리해주시면 다른 사람이 코드를 읽는데 많은 도움이 됩니다.

GangWoon commented on Sidedish/Sidedish/View/BadgeLabel.swift:
```diff
@@ -0,0 +1,17 @@
+class BadgeLabel: UILabel {
+
+    override func layoutSubviews() {
+        self.layer.cornerRadius = 8
```
layoutSubviews에서 작성하신 이유가 있을까요?

lenaios replied:
layoutSubviews()가 호출될 때 뷰가 다시 그려지는 흐름을 생각하고 작성했습니다만, 흐름상 많이 어색할까요?

GangWoon replied:
흐름이라는 감정적인 접근보다 코드에 위치가 타당한지 궁금합니다. 
해당 메소드가 호출되는 횟수는 한 번 뿐인가요??

okstring replied:
여러번 호출되는 것을 간과하고 있었던 것 같습니다.
label이 생성될 때(init) 그려지도록 수정하도록 하겠습니다! :)

GangWoon commented on Sidedish/Sidedish/View/ItemCollectionViewCell.swift:
```diff
@@ -0,0 +1,61 @@
+        dishDescription.text = model.description
+        sellingPrice.text = model.sPrice
+        
+        let priceString = model.nPrice == nil ? "" : "\(model.nPrice ?? "")원"
```
데이터는 뷰가 아닌 다른 곳에서 변경하는게 어떨까요?

GangWoon commented on Sidedish/Sidedish/View/ItemCollectionViewCell.swift:
```diff
@@ -0,0 +1,61 @@
+        attributeString.addAttribute(NSAttributedString.Key.strikethroughStyle, value: 2, range: NSMakeRange(0, attributeString.length))
+        originalPrice.attributedText = attributeString
+        
+        model.badge?.forEach { text in
```
뷰에서 로직을 처리하는 게 아닌 다른 곳에서 처리하는 건 어떨까요?

GangWoon commented on Sidedish/Sidedish/View/ItemCollectionViewCell.swift:
```diff
@@ -0,0 +1,61 @@
+    }
+}
+
+enum Badge: String {
```
정보: enum 타입 rawValue를 사용하는 방법 이외에도 CustomStringConvertible를 채택해서 사용하는 방법도 있습니다.

GangWoon commented on Sidedish/Sidedish/View/TapToastGestureRecognizer.swift:
```diff
@@ -0,0 +1,19 @@
+    var countText: String
+    
+    override init(target: Any?, action: Selector?) {
+        self.title = ""
```
title, countText를 초기화 단계에서 받는 건 어떨까요?

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,105 @@
+    var itemViewModel: ItemViewModel!
+    var headerViewModel: HeaderViewModel!
+    var fetchItemSubscription = Set<AnyCancellable>()
+    var fetchImageSubscription = Set<AnyCancellable>()
```
fetchItemSubscription, fetchImageSubscription를 생성하신 이유가 있을까요?

lenaios replied:
Combine을 활용해서 뷰와 뷰모델을 연결해주고 있는데, 관련해서 뷰모델이 변경되었을 때 데이터를 저장하기 위해 필요한 값으로 이해하고 작성했습니다. combine 사용이 아직 미숙해서 조금 더 공부 해보겠습니다!

GangWoon replied:
제가 질문을 잘 못 한 거 같네요.
두 개의 인스턴스를 생성하신 이유가 있을까요? 
서로 다른 작업을 하는 친구들인가요?

okstring replied:
각자 다른 작업을 하게끔 나눠서 처리하도록 했습니다.(이미지 정보와 그 외 item정보들을 분리해서 처리)
 - 둘을 나눈 이유는 image를 lazy하게 처리하기 위해 고민해본 결과 먼저 item 정보를 요청하고 image 정보를 요청하고 있습니다

okstring replied:
밑에서 리뷰해주신 부분을 고치고 있는 와중에 깨달았습니다! 
`fetchImageSubscription `가 굳이 필요 없을 것 같습니다 😅

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,105 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        collectionView.delegate = self
+        collectionView.dataSource = self
```
참고: 스토리보드에서 델리게이트를 설정할 수 있는 방법이 있습니다.

lenaios replied:
궁금점: 현업에서는 스토리보드에서 설정하는 것을 지향하는 편인가요?🤔

GangWoon replied:
회사마다 다른 거 같습니다 ㅋㅋ

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,105 @@
+        self.itemViewModel.$items
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.collectionView.reloadSections(IndexSet(integer: 0))
```
collectionView.reloadData와 다른 점이 있을까요?

lenaios replied:
collectionView.reloadData는 collectionView 전체를 다시 리로드 하는 것이고,
reloadSections는 해당하는 section을 리로드 하는 차이가 있는 것으로 알고 있습니다.
reloadSections을 사용한 이유는 collectionView.reloadData보다 비용이 적다고 생각해서입니다!

GangWoon replied:
말씀하신 내용은 맞지만 현재 코드에서도 동일한 결과를 예상할 수 있을까요?

GangWoon replied:
이미지가 다운받을 때마다 섹션을 리로드하게 되는데 자원적인 측면에 낭비가 발생하고 있지 않을까요?

okstring replied:
쉬운 방법을 찾고 있지 않았나 생각이 들었습니다.
이미지를 하나씩 reload하도록 수정하겠습니다! :)

GangWoon replied:
제 말은 10개의 이미지를 다운 받게되면, collectionView가 10번 섹션을 reload한다는 말이 였습니다.

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,105 @@
+        let item = self.itemViewModel.items[indexPath.row]
+        
+        cell.configure(model: item)
+        guard let data = self.itemViewModel.images[indexPath.row] else { return cell }
```
이미지가 없을 경우에 리턴시키는게 맞을까요?

lenaios replied:
이미지가 없더라도, 상품명과 설명 등 다른 데이터라도 보여주는 것이 맞다고 생각해서 cell을 리턴시켜주도록 했는데
빈 이미지를 보여주기 보다는 기본 이미지를 보여주는 것이 좋을 것 같다는 생각이 듭니다.
이를 의도하신게 맞을까요?👀

GangWoon replied:
UIimageview.image는 타입이 어떻게 될까요? 가드문 정말 필요한지 궁금합니다.

okstring replied:
> `guard let data = self.itemViewModel.images[indexPath.row] else { return cell }`

이 로직이 필요한 이유는 url로 이미지 Data를 요청했을 때 데이터가 안오는 경우도 생각했고 
각 item의 배열 수만큼 `ItemViewModel.images` 도 동일한 크기(count)를 가지는 배열을 만들어줘서 
해당 배열의 index에 접근을 해 data를 가져올 때 nil 값이 있을 수도 있습니다. 

그 처리를 위해 부득이하게 guard 문을 두게 되었는데 구조를 변경하는 것이 좋을지 궁금합니다🤔

GangWoon replied:
아... 제가 착각했습니다. 혼란을 드려서 죄송합니다.

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,105 @@
+        let header = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionView.elementKindSectionHeader,
+                                                                     withReuseIdentifier: HeaderCollectionReusableView.identifier,
+                                                                     for: indexPath) as! HeaderCollectionReusableView
+        let tapRecognizer = TapToastGestureRecognize(target: self, action: #selector(makingHeaderToast(_:)))
```
UITapGestureRecognizer를 커스터마이징 하지 않고 할 수 있는 방법은 없을까요?

lenaios replied:
headerView에서 제스쳐를 인식하기 위해 제스처를 사용했는데 마땅한 방법이 떠오르지 않아 서브클래싱을 사용하게 됐습니다.
1. 생각해낸 다른 방법으로는 tag를 쓰는 방법인데 title, countLabel에 해당하는 정보를 tag에 표현할 수 없고 사용을 지양하고 있다고 배워서 적절하지 않다고 느꼈고
2. notification를 쓰는 방법은 view-controller사이에서 쓰는 것은 어색하다고 했던 의견을 이전에 받아서 사용이 어려웠던 것 같습니다.
3. 마지막 방법으로는 title에 대한 부분을 button으로 바꿔 진행하는 것이 좋을 것이라 생각했습니다 :) 

혹시 다른 방법이있을지 키워드를 주시면 감사하겠습니다! :)

GangWoon replied:
제가 해당 코드 부분을 읽었을 때 느낀점은 addTarget에서 파라미터로 데이터를 받을 수 없기 때문에 받을 수 있는 인터페이스를 옮기신 거 같았어요. 맞으신가요? 

위와 같은 문제를 만나셨다면 클로저를 생성해서 처리할 수 있지 않을까요?

okstring replied:
정말 죄송하지만 오랜시간 고민을 해보고 여러 시도를 해봤지만 클로저를 사용하는 방법으로 해결을 못했습니다..! 클로저가 선언되어있는 위치가 어딘지 감을 못잡고 있습니다 😭 조금만 더 키워드를 부탁드릴 수 있을까요..?
검색해본 키워드 : `UIView action closure`, `UITapGesture`, `selector parameter`, `swift selector store property`

GangWoon replied:
앞의 문구는 빼주세요 ㅋㅋㅋ 
viewController 내부에서 
`var headerViewActionHandler: (() -> Void)?`를 프로퍼티로 갖고 있고,

viewDidLoad에서
```
        headerViewActionHandler = {  
            Toast(text: self.viewModel.item").show()
        }
```

@objc func didTappedSectionHeader() {
headerViewActionHandler()
}

이런식으로 분리하게 되면 addTarget()에서 클로저를 넘길 때 파라미터에 데이터를 집어넣지 못하더라도 해결할 수 있지 않을까요?

lenaios committed:
CollectionViewCell에서 데이터 가공하는 로직을 ViewController로 이동

okstring committed:
Set: 피드백 반영

 - CustomStringConvertible 활용
 - NSLog 고려
 - tapGestureRecognize convenience init 사용
 - delegate, dataSource storyboard에서 설정

okstring commented:
추가로 궁금한 부분은 제외하고 말씀주신 피드백 반영해서 push 완료했습니다!
감사합니다 :)

okstring committed:
2차 피드백 반영

 - DEBUG시 error 출력
 - badge label init 시 radius 설정
 - section이 아닌 cell 별로 reload

okstring committed:
3차 피드백 반영

 - imageData 프로퍼티 추가, 필요없는 속성 제거
 - 각 부분 protocol 채택
 - Networking 제거
 - Gesture 서브클래싱 제거 후 핸들러 설정

okstring commented:
오전에 피드백 주신 내용 수정 완료했습니다!
지금은 여러 고민해봤지만 reload하는 시점이 해결되지가 않았고(main쓰레드에서 load시 런타임 에러) 또 세션을 추가하게되면 구조가 달라질 것 같아 너무 지연이 될 것 같은 판단이 들었습니다.
일단 머지를 받고 다음 PR때 reload부분 보완하려 계획중에 있습니다
감사합니다 :)

GangWoon commented on Sidedish/Sidedish/Entity/Enum/Badge.swift:
```diff
@@ -0,0 +1,17 @@
+    case launch = "론칭특가"
+    
+    var description: String {
+        return self.rawValue
```
rawValue를 사용하시는 것보다 
description 내부에서 switch문으로 분류하는 게 더 안전해보입니다.

GangWoon commented on Sidedish/Sidedish/Entity/SidedishItem.swift:
```diff
@@ -0,0 +1,33 @@
+
+struct SidedishItem: Codable {
+    var detailHash: String
+    var image: String
```
image 타입이 string인데 뜻하는게 url에 더 가깝지 않을까요?

GangWoon commented on Sidedish/Sidedish/Repository/SidedishNetworkCenter.swift:
```diff
@@ -0,0 +1,56 @@
+
+class SidedishNetworkCenter: Networkable {
+    func fetchItems(url: String, completion: @escaping (Result<[SidedishItem], AFError>) -> ()) {
+        //MARK: - cache가 없으면
```
주석은 정말 필요로할 때 보통 사용합니다. 
ex) iOS 14.0 부턴 api가 변경됨으로, 최소 버전이 14.0일 때 개선해주세요.
메소드 네이밍이 정말 어려운 문제지만 해당 메소드가 무슨일을 하는지 표현하는 방법도 연습하셨으면 좋겠습니다.

GangWoon commented on Sidedish/Sidedish/View/TapToastGestureRecognizer.swift:
```diff
@@ -0,0 +1,25 @@
+
+import UIKit
+
+class TapToastGestureRecognize: UITapGestureRecognizer {
```
코드를 변경한 지금 TapToastGestureRecognize가 필요한가요?

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let networking = SidedishNetworkCenter()
```
뷰에서 모든 디펜던시를 생성하는 건 옳지 않습니다.
앞써 말씀드린대로 각 영역마다의 독립성을 보존하는게 정말 중요한 개념입니다.
viewModel로 옮겨지는 게 좋을 거 같습니다.

GangWoon commented on Sidedish/Sidedish/ViewController.swift:
```diff
@@ -0,0 +1,124 @@
+            return ItemCollectionViewCell()
+        }
+        let item = self.itemViewModel.items[indexPath.row]
+        let priceString = item.nPrice == nil ? "" : "\(item.nPrice ?? "")원"
```
data + "원" 이런 형태의 모든 부분에 뷰에서 작업하는 것보다 viewModel에서 만들어주고 뷰에선 할당만 하는 형식이 어떤가요? 라고 말씀드렸던 겁니다.

GangWoon reviewed (approved):
@okstring  @lenaios 
피알을 수정하기 앞써 이번에 작성한 코맨트는 무시하시고 프로젝트를 진행하셔도 좋습니다.
더 이상 고칠게 없다고 판단하시면 스쿼시 머지로 부탁드리겠습니다.

코드를 보개되면 특정 부분에선 다음 줄을 공백으로 두고 넘어가거나 어떤 부분은 띄웠고 어떤 부분은 붙이는 식으로 코드가 작성되어있습니다.
현업에서 모든 코드가 하나의 컨밴션을 준수하는 건 아니지만 개인적으로 협업에 있어서 중요한 부분이라고 생각합니다. 앞으로 코드를 작성할 때 두분이서 의논하셔서 코드 컨밴션을 맞춰주셨으면 좋겠습니다.

클린 아키텍처를 반영하는게 많이 어려우셨을 텐데 다들 고생하셨습니다 :)

lenaios committed:
4차 피드백 반영

- 변수 네이밍 변경
- 불필요한 주석, 파일 제거
- data + "원" 형태 처리를 viewModel에서 하도록 수정
